docs/building-blocks/extending-templates/objects-as-parameters.md

```diff
@@ -0,0 +1,326 @@
+                "parameters": {},
+                "variables": {},
+                "resources": [],
+                "outputs": {}
+            }
+        }       
+    },
+    {
+        "apiVersion": "2015-01-01",
+        "type": "Microsoft.Resources/deployments",
+        "name": "[concat('loop-', copyIndex(1))]",
+        "dependsOn": [
+          "[concat('loop-', copyIndex())]"
+        ],
+        "copy": {
+          "name": "iterator",
+          "count": "[length(parameters('networkSecurityGroupsSettings').securityRules)]"
+        },
+        "properties": {
+          "mode": "Incremental",
+          "template": {
+            "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+            "contentVersion": "1.0.0.0",
+           "parameters": {},
+            "variables": {},
+            "resources": [
+                {
+                    "name": "[concat('NSG1/' , parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].name)]",
+                    "type": "Microsoft.Network/networkSecurityGroups/securityRules",
+                    "apiVersion": "2016-09-01",
+                    "location":"[resourceGroup().location]",
+                    "properties":{
+                        "description": "[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].description]",
+                        "priority":"[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].priority]",
+                        "protocol":"[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].protocol]",
+                        "sourcePortRange": "[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].sourcePortRange]",
+                        "destinationPortRange": "[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].destinationPortRange]",
+                        "sourceAddressPrefix": "[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].sourceAddressPrefix]",
+                        "destinationAddressPrefix": "[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].destinationAddressPrefix]",
+                        "access":"[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].access]",
+                        "direction":"[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].direction]"
+                        }
+                  }
+            ],
+            "outputs": {}
+          }
+        }
+    }
+  ],          
+  "outputs": {}
+}
+
+```
+
+Let's take a closer look at how we specify our property values in the `"securityRules"` child resource: all of our properties are referenced using the `parameter()` function, and then we use the dot operator `.` to reference our `"securityRules"` array, indexed by the current value of the iteration. Finally, we use another dot operator `.` to reference the name of the object. 
```

Comment:
I think just "dot operator" is clear, you may not need the `.`. (Elsewhere you just say "dot operator" which seemed fine)